src/com/amazon/ion/impl/IonReaderLookaheadBuffer.java

```diff
@@ -544,9 +546,9 @@ private void reclaimNopPadding() {
      * @return the number of bytes actually skipped.
      * @throws IOException if thrown by the underlying InputStream.
      */
-    private int skipBytesFromInput(int numberOfBytesToSkip) throws IOException {
+    private long skipBytesFromInput(long numberOfBytesToSkip) throws IOException {
         try {
-            return (int) getInput().skip(numberOfBytesToSkip);
+            return getInput().skip(numberOfBytesToSkip);
```

Comment:
Same note RE: `skip`'s return value.

Reply:
The IonReaderLookaheadBuffer handles this case. It yields back to the reader for more data as soon as skip returns < 1 ([here](https://github.com/amzn/ion-java/blob/master/src/com/amazon/ion/impl/IonReaderLookaheadBuffer.java#L746) and [here](https://github.com/amzn/ion-java/blob/master/src/com/amazon/ion/impl/IonReaderLookaheadBuffer.java#L792)).